theano/gof/link.py

```diff
@@ -496,7 +496,10 @@ def map_storage(fgraph, order, input_storage, output_storage):
     else:
         assert len(fgraph.inputs) == len(input_storage)
 
-    storage_map = {}
+    if storage_map is None:
+        storage_map = {}
+
+    # add input storage into storage_map
     for r, storage in zip(fgraph.inputs, input_storage):
         storage_map[r] = storage
```

Comment:
This with the line 495 cause that if the storage_map is provided, but not input_storage, we will change the storage_map of inputs to [None]. Do we want that? If not, you can change line 495 to use the content of the storage_map if provided.

If both, storage_map and input_storage and provided, assert that they point to the same content. If not, raise an error.